app/models/playlist.rb

```diff
@@ -125,27 +126,34 @@ def self.latest(limit = 5)
   end
 
   def ensure_private_if_less_than_two_tracks
-    self.private = true if !is_favorite? && (tracks_count < 2)
+    self.private = true if !is_favorite? && !can_be_public?
     true
   end
 
-  # playlist is a mix if there is at least one track with a track from another user
-  def set_mix_or_album
-    # is this a favorites playlist?
-    is_mix = true if is_favorite?
-    is_mix = true if tracks.present? && tracks.count > tracks.where('user_id != ?', user&.id).count
-    true
+  # list any required conditions before playlist
+  # can be made public
+  def can_be_public?
+    tracks_count >= 2
   end
 
   # if this is a "favorites" playlist, give it a name/description to match
   def name_favorites_and_set_permalink
     self.title = user.name + "'s favorite tracks" if is_favorite?
+    # move me to new tracks_controller#create
+    self.is_mix = true if consider_a_mix?
+
     generate_permalink!
   end
 
   def check_for_new_permalink
     generate_permalink! if title_changed?
   end
+
+  def consider_a_mix?
+    return true if is_favorite?
+
+    tracks.present? && tracks.count > tracks.where('user_id != ?', user&.id).count
```

Comment:
Something like `tracks.pluck(:user_id).uniq > 1` might be an upgrade here, more direct and less queries.

Reply:
Oh this is such a sweet trick. I love it.